internal/ovirt/validation.go

```diff
@@ -315,3 +315,28 @@ func validateEnum(values []string) schema.SchemaValidateDiagFunc {
 		}
 	}
 }
+
+func validateHugePages(i interface{}, path cty.Path) diag.Diagnostics {
+	val, ok := i.(int)
+	if !ok {
+		return diag.Diagnostics{
+			diag.Diagnostic{
+				Severity:      diag.Error,
+				Summary:       "Huge pages must be an integer.",
+				Detail:        "The provided huge pages is not an integer.",
+				AttributePath: path,
+			},
+		}
+	}
+	if ovirtclient.VMHugePages(val).Validate() == nil {
```

Comment:
Why not just use the error returned from the Validate() function in the returned diag structure?